opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,111 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=protected-access
+
+from logging import getLogger
+from re import match
```

Comment:
Well, there is another option, not support it at all. It is optional and not well defined, we can decide no to support it until it is.

If we decide to support it, I would use a regex, it is easier to document the behavior (just tell the users we are using regular regexes) and regexes are pretty much a superset of features that pretty much includes "wildcards".

Reply:
> The only issue with leaving it out from our initial release is that it would be backward incompatible. For example, if we start with exact match only and user has `instrument_name=http.bar`, if we add regexes it will now match also `http_bar`. Not a huge concern though.
> 
> I agree regexes would be more powerful, but just want to call out the spec examples are like `http.*` vs `http\..*`. I think we should clarify this with the Spec SIG. IIRC, one of the reasons they went with simpler wildcard matching is to have it be uniform across all language SDKs, while regex flavors vary in different languages.

I can bring this topic in the next spec SIG, I agree that is the place where to discuss it. I don't agree completely with their argument in favor of wildcards to make it more uniform across all languages, a Python View will only be used with a Python interpreter and standard library, and that view already has many other Python-specific details besides the regex/wildcard syntax. So In my opinion, there is not much to gain in trying to use a uniform regex/wildcard syntax among languages.

Reply:
Discussed in the SIG, this is the [result](https://github.com/open-telemetry/opentelemetry-specification/pull/2325).